### DISCUSSION TITLE
sleep method inside pocketbase

### DISCUSSION THREAD

**Discussion by nikhil-varma-mudunuri** [pocketbase/pocketbase#4001] [Q&A] [answered]
q1) is there a way to use some equivalent of sleep inside pocketbase extend with js
q2) when i use create record operation in frontend , and at backend (extend with js) i have written some code inside onModelBeforeCreate, and if take some 30 secs to run that code, will frontend wait that much time when we use await function

**Comment by ganigeorgiev** [comment, marked as answer]
1. The JS hooks doesn't expose `time.Sleep` or other similar Go method at the moment.

2. Yes, it will block. If you need to execute something within a hook in a non-blocking manner you'll need to use Go and call your function in a new goroutine.

**Comment by nikhil-varma-mudunuri** [comment]
oh, thanks, for question 2 i actually needed it to be blocked only.
